src/components/dialog/dialog.js

```diff
@@ -151,45 +152,15 @@ class Dialog extends Modal {
    */
   get dialogTitle() {
     if (this.props.title) {
-      return <h2 className={ this.dialogTitleClasses } data-element='title'>{ this.props.title }</h2>;
-    }
-
-    return null;
-  }
-
-  /**
-   * Returns HTML and text for the dialog subtitle.
-   *
-   * @method dialogSubtitle
-   * @return {String} subtitle to display
-   */
-  get dialogSubtitle() {
-    if (this.props.subtitle) {
-      return <p className={ this.dialogSubtitleClasses } data-element='subtitle'>{ this.props.subtitle }</p>;
+      return (
+        <Heading title={ this.props.title } subheader={ this.props.subtitle } />
```

Comment:
it is added by the heading component https://github.com/Sage/carbon/blob/master/src/components/heading/heading.js#L198
it is already tested there https://github.com/Sage/carbon/blob/master/src/components/dialog/__spec__.js#L325